### PR TITLE
Thorough request object retrieval algorithm

### PR DESCRIPTION
**Pull request by acrusage**
Improved algorithm for retrieving AuthenticationResponseParameters as asked for in https://github.com/a-sit-plus/compose-wallet-app/pull/61#discussion_r1534006344

### CODE REVIEW

**Review by JesusMcCloud — CHANGES_REQUESTED** [review]
The implementation looks really good, but it would be nice to have both cases tested and not only the baseline one we hade before (even though it is important to keep it as regression test)

**Comment by acrusage** [comment]
> The implementation looks really good, but it would be nice to have both cases tested and not only the baseline one we hade before (even though it is important to keep it as regression test)

I might need help with that, I'm not sure how to provide a dummy response to a HttpClient.get request

**Comment by acrusage** [comment]
> The implementation looks really good, but it would be nice to have both cases tested and not only the baseline one we hade before (even though it is important to keep it as regression test)

fixed in commit 19b461408a25dfa5d43a8970ea973d2ee4db621b

**Review by JesusMcCloud — APPROVED** [review]

**Comment by nodh** [comment]
I'm unhappy with the added dependency on `HttpClient` from  ktor ... is this really necessary?

**Comment by JesusMcCloud** [comment]
how would you make an http request, without an http client?

File: vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt [hunk 1]
```diff
@@ -386,3 +465,17 @@ class OidcSiopWallet(
     }
 
 }
+
+typealias RequestObjectCandidateRetriever = suspend (Url) -> List<String>
+
+private val HttpClient.asRequestObjectCandidateRetriever: RequestObjectCandidateRetriever
```
Comment (nodh) [inline vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt:471]: imho `as` as the prefix indicates a function, but this is a property

**Comment by acrusage** [comment]
> > how would you make an http request, without an http client?
> 
> Why don't we leave that up to the client, i.e. the caller of our library, as it has been the case up until now?

fixed in commit 03a0cdd1303f7c2e0e17cbc4d665635566875f47 
prepared at pull request https://github.com/a-sit-plus/kmm-vc-library/pull/46 along with a fix for https://github.com/a-sit-plus/kmm-vc-library/pull/45#discussion_r1547342169

**Pull request merged**